src/http_server.cc

```diff
@@ -2443,6 +2453,14 @@ HTTPAPIServer::HandleCudaSharedMemory(
           }
 
           if (err == nullptr) {
+            if (b64_handle_len > INT_MAX) {
```

Comment:
Can we create a wrapper function on top of `base64_decode_block`? It can take the string_len as size_t, perform this check on the length, and pass it to the `base64_decode_block` as const int. 
All the setup logic like base64_init_decodestate can be part of this wrapper too.

This change will enforce DRY by keeping the check and decode logic in a single place and hence making the code more maintainable.